### PR TITLE
Scope git calls to the current buffer path

### PR DESCRIPTION
**Pull request by robertgzr**

### CODE REVIEW

File: lua/gitlinker/git.lua [hunk 1]
```diff
@@ -1,97 +1,64 @@
...
 
 local job = require("plenary.job")
+local path = require("plenary.path")
 
-local function get_remotes()
-  local remotes
-  local p = job:new({ command = "git", args = { "remote" } })
+-- wrap the git command to do the rigth thing alwasy
+-- @args: can be any git subcommand
+local function git(args)
```
Comment (ruifm) [inline lua/gitlinker/git.lua:8]: This comment is spurious and can be removed, I think the parameter `args` in a function called `git` is very self-explanatory.

Besides, there is no need for formal docs in local non-public functions!

File: lua/gitlinker/git.lua [hunk 2]
```diff
@@ -1,97 +1,64 @@
...
+  local o = { '-C', tostring(path:new(vim.api.nvim_buf_get_name(0)):parent()) }
+  vim.list_extend(o, args)
```
Comment (ruifm) [inline lua/gitlinker/git.lua:10]: sorry I'm picky with variable names but `o` needs to be renamed to something like `scope_args`. Either that or you can just do:

```suggestion
  vim.list_extend(args, { '-C', tostring(path:new(vim.api.nvim_buf_get_name(0)):parent()) })
```

File: lua/gitlinker/git.lua [hunk 3]
```diff
@@ -1,97 +1,64 @@
...
-    args = { "diff", "--exit-code", rev, "--", file },
-  })
-  p:after_failure(function()
-    has_changed = true
-  end)
-  p:sync()
-  return has_changed
+  if git({ "diff", "--exit-code", rev, "--", file }) then
+    return false
+  end
```
Comment (ruifm) [inline lua/gitlinker/git.lua:48]: there is no longer need for `--exit-code` since it's now unused (which is nice, less code :+1: )

**Review by ruifm — APPROVED** [review]
Amazing refactor, many thanks for this :1st_place_medal:

Just a couple of review nits. Also, could you preface the commit message with `refactor: `?

**Comment by ruifm** [comment]
@robertgzr just a question: does this scoping arg fix any issue you had?

**Comment by robertgzr** [comment]
@ruifm sorry for the silence on this one. I'll give some attention to these prs over the holidays

> just a question: does this scoping arg fix any issue you had?

I often open vim in a non-git dir or edit files from multiple repos. The git tool will always we called from vim's current working directory though, which is not the behavior I had expected.

**Comment by robertgzr** [comment]
I think I responded to all the comments. I'll squash when you had a chance to look :)

**Pull request merged**